src/main/java/org/dcsa/ebl/model/Booking.java

```diff
@@ -1,103 +1,104 @@
-    @JsonProperty("carrierBookingReference")
+
+    @Id
     @Column("carrier_booking_reference")
+    @JsonProperty("carrierBookingReference")
     @Size(max = 35)
-    @Id
     private String id;
 
     @Column("service_type_at_origin")
-    @Size(max = 5)
-    @NotNull
+    @Size(max = 3)
     private ServiceType serviceTypeAtOrigin;
-    
-    public void setServiceTypeAtOrigin(@NotNull String serviceTypeAtOrigin) {
+
+    public void setServiceTypeAtOrigin(String serviceTypeAtOrigin) {
         this.serviceTypeAtOrigin = ServiceType.valueOf(serviceTypeAtOrigin);
     }
 
-    public void setServiceTypeAtOrigin(@NotNull ServiceType serviceTypeAtOrigin) {
+    public void setServiceTypeAtOrigin(ServiceType serviceTypeAtOrigin) {
         this.serviceTypeAtOrigin = serviceTypeAtOrigin;
     }
 
     @Column("service_type_at_destination")
-    @Size(max = 5)
-    @NotNull
+    @Size(max = 3)
     private ServiceType serviceTypeAtDestination;
 
-    public void setServiceTypeAtDestination(@NotNull String serviceTypeAtDestination) {
+    public void setServiceTypeAtDestination(String serviceTypeAtDestination) {
         this.serviceTypeAtDestination = ServiceType.valueOf(serviceTypeAtDestination);
     }
 
-    public void setServiceTypeAtDestination(@NotNull ServiceType serviceTypeAtDestination) {
+    public void setServiceTypeAtDestination(ServiceType serviceTypeAtDestination) {
         this.serviceTypeAtDestination = serviceTypeAtDestination;
     }
 
     @Column("shipment_term_at_origin")
-    @Size(max = 5)
-    @NotNull
+    @Size(max = 3)
     private ShipmentTerm shipmentTermAtOrigin;
 
-    public void setShipmentTermAtOrigin(@NotNull String shipmentTermAtOrigin) {
+    public void setShipmentTermAtOrigin(String shipmentTermAtOrigin) {
         this.shipmentTermAtOrigin = ShipmentTerm.valueOf(shipmentTermAtOrigin);
     }
 
-    public void setShipmentTermAtOrigin(@NotNull ShipmentTerm shipmentTermAtOrigin) {
+    public void setShipmentTermAtOrigin(ShipmentTerm shipmentTermAtOrigin) {
         this.shipmentTermAtOrigin = shipmentTermAtOrigin;
     }
 
     @Column("shipment_term_at_destination")
-    @Size(max = 5)
-    @NotNull
+    @Size(max = 3)
     private ShipmentTerm shipmentTermAtDestination;
 
-    public void setShipmentTermAtDestination(@NotNull String shipmentTermAtDestination) {
+    public void setShipmentTermAtDestination(String shipmentTermAtDestination) {
         this.shipmentTermAtDestination = ShipmentTerm.valueOf(shipmentTermAtDestination);
     }
 
-    public void setShipmentTermAtDestination(@NotNull ShipmentTerm shipmentTermAtDestination) {
+    public void setShipmentTermAtDestination(ShipmentTerm shipmentTermAtDestination) {
         this.shipmentTermAtDestination = shipmentTermAtDestination;
     }
 
     @Column("booking_datetime")
-    @NotNull
-    private LocalDateTime bookingDatetime;
+    @JsonFormat(shape = JsonFormat.Shape.STRING)
+    private LocalDateTime bookingDateTime;
```

Comment:
How come this is not a ZonedDateTime? LocalDateTime does not make sense in an international setting (which EBL is in) unless there is a point of reference for the LocalDateTime (at which point you have an "indirect" ZonedDateTime).

Related, I would probably remove the @JsonFormat and use the default date format (unless we have a pressing reason to not use the default, which is ISO-8601 as I recall)